src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0017Tests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Xunit;
+using Verifier = Azure.ClientSdk.Analyzers.Tests.AzureAnalyzerVerifier<Azure.ClientSdk.Analyzers.ClientOptionsAnalyzer>;
+
+namespace Azure.ClientSdk.Analyzers.Tests
+{
+    public class AZC0017Tests
+    {
+        [Fact]
+        public async Task AZC0017ProducedForServiceVersionWithNoncapitalizedEnumName()
+        {
+            const string code = @"
+namespace RandomNamespace
+{
+    public class SomeClientOptions
+    {
+        public enum ServiceVersion
+        {
+            v2021_05_01 = 1,
```

Comment:
Can you use a marker syntax here? It might be easier than keeping track of the exact span.

Reply:
Could you clarify? This is my first foray into Roslyn analyzers. I would add that I considered ways to span *just* the typo'd word/part, but taking a dependency on a new package for the `StringTokenizer` that gave me spans when split or writing my own split-to-spans seemed overkill given the target audience / use case.

Reply:
https://github.com/Azure/azure-sdk-tools/blob/main/src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0013Tests.cs#L21

The `[|....|]` part defines a span for an expected diagnostic. This way you don't need to manually figure out the location.

Reply:
Oh, that's what that was! I saw that in a few places and wasn't sure what it was supposed to be.